### PR TITLE
collect CPU info with collect_env.py for new issues reporting

### PR DESCRIPTION
**Pull request by jingxu10**
Add CPU information collection feature to collect_env.py for new issues reporting. This helps us to triage issues on CPU.

### CODE REVIEW

**Comment by jingxu10** [comment]
@malfet @jgong5

File: torch/utils/collect_env.py [hunk 1]
```diff
@@ -203,6 +204,23 @@ def get_nvidia_smi():
     return smi
 
 
+def get_cpu_info(run_lambda):
+    rc = 0
+    out = ''
+    err = ''
+    if get_platform() == 'linux':
+        rc, out, err = run_lambda('lscpu')
```
Comment (malfet) [inline torch/utils/collect_env.py:212]: Why not `cat /proc/cpuinfo` or use cpuinfo provided by PyTorch?

File: torch/utils/collect_env.py [hunk 2]
```diff
@@ -203,6 +204,23 @@ def get_nvidia_smi():
...
+    elif get_platform() == 'win32':
+        rc, out, err = run_lambda('wmic cpu get Name,Manufacturer,Family,Architecture,ProcessorType,DeviceID,\
+        CurrentClockSpeed,MaxClockSpeed,L2CacheSize,L2CacheSpeed,Revision /VALUE')
```
Comment (malfet) [inline torch/utils/collect_env.py:215]:
```suggestion
        CurrentClockSpeed,MaxClockSpeed,L2CacheSize,L2CacheSpeed,Revision /VALUE')
    elif get_platform() == 'darwin':
        rc, out, err = run_lambda("sysctl -n machdep.cpu.brand_string")
```

**Review by malfet — APPROVED** [review]
Looks fine, but please include example outputs for different OSes(also, do not include MacOS)

Also, why expect on `lscpu`(which might not be installed), rather than grep some information from `/proc/cpuinfo`, which is always available?

File: torch/utils/collect_env.py (hunk 1 above)
Comment (jingxu10) [reply torch/utils/collect_env.py:210]: good point. I'm looking into this.

Comment (jingxu10) [reply torch/utils/collect_env.py:210]: `cpuinfo` dumps info of all individual cores. It is too detailed. What we need is aggregated info in high level provided by `lscpu`, like number of cores, how cores are distributed across numa nodes.

**Comment by jingxu10** [comment]
Hi @malfet , I updated the example output to the script. Also, the reason of using `lscpu` is as below. I commented in your original comment.
`cpuinfo` dumps info of all individual cores. It is too detailed. What we need is aggregated info in high level provided by `lscpu`, like number of cores, how cores are distributed across numa nodes.

I tried to start merge, but it failed due to some reasons. If the PR looks good, would you help to merge it? Thank you.

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**